### PR TITLE
docs(settings): bunch of fixes and improvements

### PR DESCRIPTION
**Pull request by mkniewallner**
## Summary

Spotted some issues in the settings documentation, and room for small improvements by linking to PEPs/RFCs.

Also updating contribution documentation to mention that it's necessary to enable the virtual environment before running `mkdocs serve`.

## Test Plan

Running documentation locally.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -127,11 +127,21 @@ To preview any changes to the documentation locally:
...
+
+    ```shell
+    # On macOS and Linux.
+    source .venv/bin/activate
+
+    # On Windows.
+    .venv\Scripts\activate
+    ```
+
 1. Run the development server with:
```
Comment (CoolCat467) [inline CONTRIBUTING.md:145]: The bullet point numbers below this are now off

**Review by CoolCat467 — CHANGES_REQUESTED** [review]
Looks pretty good other than a small thing I noticed

File: CONTRIBUTING.md (hunk 1 above)
Comment (mkniewallner) [reply CONTRIBUTING.md:145]: In markdown, it doesn't matter which number you use for ordered lists, as using "1" for multiple consecutive items will still generate a numerically ordered list: https://www.markdownguide.org/basic-syntax/#ordered-lists

Comment (CoolCat467) [reply CONTRIBUTING.md:145]: Interesting, didn't know about that feature!

**Review by charliermarsh — APPROVED** [review]
Excellent, thank you!

**Pull request merged**